app/scripts/services/templates.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .service("CachedTemplateService", function(){
+
+    var template = null;
+    return {
+      setTemplate: function(temp) {
+        template = temp;
+      },
+      getTemplate: function() {
+        return template;
+      },
+      clearTemplate: function() {
+        template = null;
+      }
+    };
+
+  }).service("ProcessedParametersService", function(){
+
+    var params = {
+      generated: [],
```

Comment:
hmm these don't seem right, really its
params.all and params.generated

i was hoping it would have been a single array with generated being a flag on the individual params, but i see where that would complicate you changing from the one generated param to many generated params sentences in the view :-\